pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -70,59 +83,182 @@ export class PyTerminalPlugin extends Plugin {
+    abstract stderr_writeline(msg: string): void;
+}
+
+function make_PyTerminal_pre(app: PyScriptApp) {
     /** The <py-terminal> custom element, which automatically register a stdio
      *  listener to capture and display stdout/stderr
      */
-    class PyTerminal extends HTMLElement implements Stdio {
+    class PyTerminalPre extends PyTerminalBaseClass {
         outElem: HTMLElement;
-        autoShowOnNextLine: boolean;
 
         connectedCallback() {
             // should we use a shadowRoot instead? It looks unnecessarily
             // complicated to me, but I'm not really sure about the
             // implications
             this.outElem = document.createElement('pre');
-            this.outElem.className = 'py-terminal';
+            this.outElem.classList.add('py-terminal');
             this.appendChild(this.outElem);
 
-            if (this.isAuto()) {
-                this.classList.add('py-terminal-hidden');
-                this.autoShowOnNextLine = true;
-            } else {
-                this.autoShowOnNextLine = false;
-            }
+            this.setupPosition(app);
+        }
 
+        // implementation of the Stdio interface
+        stdout_writeline(msg: string) {
+            this.outElem.innerText += msg + '\n';
             if (this.isDocked()) {
-                this.classList.add('py-terminal-docked');
+                this.scrollTop = this.scrollHeight;
+            }
+            if (this.autoShowOnNextLine) {
+                this.classList.remove('py-terminal-hidden');
+                this.autoShowOnNextLine = false;
             }
+        }
 
-            logger.info('Registering stdio listener');
-            app.registerStdioListener(this);
+        stderr_writeline(msg: string) {
+            this.stdout_writeline(msg);
         }
+        // end of the Stdio interface
+    }
+
+    return PyTerminalPre;
+}
+
+declare const Terminal: typeof TerminalType;
+
+function make_PyTerminal_xterm(app: PyScriptApp) {
+    /** The <py-terminal> custom element, which automatically register a stdio
+     *  listener to capture and display stdout/stderr
+     */
+    class PyTerminalXterm extends PyTerminalBaseClass {
+        outElem: HTMLDivElement;
+        _moduleResolved: boolean;
+        xtermReadyPromise: Promise<TerminalType>;
+        xterm: TerminalType;
+        cachedStdOut: Array<string>;
+        cachedStdErr: Array<string>;
+        _xterm_cdn_base_url = 'https://cdn.jsdelivr.net/npm/xterm@5.1.0';
 
-        isAuto() {
-            return this.hasAttribute('auto');
+        constructor() {
+            super();
+            this.cachedStdOut = [];
+            this.cachedStdErr = [];
+
+            // While this is false, store writes to stdout/stderr to a buffer
+            // when the xterm.js is actually ready, we will "replay" those writes
+            // and set this to true
+            this._moduleResolved = false;
+
+            //Required to make xterm appear properly
```

Comment:
Out of curiosity, is this meant to be fullscreen or just fill-up the container instead?